### PR TITLE
Add CircleCI

### PR DESCRIPTION
**Pull request by ruxandrafed**
## Overview

- Remove TravisCI and add CircleCI.
- Update Github settings to add CircleCI status checks
- Remove devDependencies and add postinstall script that install everything needed to run scripts as in CI
---

#### Meta

> Please read and confirm each of the following:

- [x] this topic was discussed in the [Technology Forum][technology-forum] _(ignore if the pull request represents small changes)_
- [x] provided a descriptive topic and overview of contribution
- [x] documentation format follows the [topic template][template]
- [x] fork is up to date _(Hint: ["Syncing a Fork"][guide-forks])_
- [x] "work in progress" commits are squashed _(Hint: ["Squashing Commits"][guide-squash])_
- [x] commits follow the [Conventional ChangeLog][conventional-changelog] format
- [x] no sensitive content included, such as:
  - content considered competitive intelligence
  - security & privacy policy violating content
  - keys, tokens or credentials

[template]: https://github.com/telus/reference-architecture/blob/master/.template.md
[conventional-changelog]: https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-angular
[guide-forks]: https://help.github.com/articles/syncing-a-fork/
[guide-squash]: https://git-scm.com/book/id/v2/Git-Tools-Rewriting-History
[technology-forum]: https://github.com/telus/technology-forum]

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      - run:
+          name: Install dependencies
+          command: npm ci
+
+jobs:
+  build:
+    executor: app
+    steps:
+      - checkout
+      - install
```
Comment (ahmadnassri) [inline circle.yml]: probably don't need an install step ..

File: package.json [hunk 2]
```diff
@@ -18,13 +18,11 @@
...
-  "devDependencies": {
-    "@telus/remark-preset-lint-markdown": "*",
-    "editorconfig-checker": "^1.3.3",
-    "remark-cli": "^6.0.0"
+    "lint:audit": "npm audit",
+    "lint:deps": "npx updated",
+    "lint:ec": "npx editorconfig-checker .",
+    "lint:md": "npx remark --quiet --frail .",
+    "lint": "npx npm-run-all -p lint:*",
+    "postinstall": "npx install-group peer --package @telus/build-essential --no-save"
```
Comment (ahmadnassri) [inline package.json]: but those already come with the docker image?

Comment (ahmadnassri) [reply package.json]: are you doing this for local development purposes?

if so, then maybe don't make this automatically trigger for people, rather use a custom script (`npm run build` maybe?), and indicate usage in the README

Comment (ruxandrafed) [reply package.json]: Yes! Will convert that to a custom script and update readme :) thanks!

**Comment by ruxandrafed** [comment]
@ahmadnassri updated, pls review :) thanks!

**Pull request merged**